crates/uv-distribution-types/src/dist_error.rs

```diff
@@ -1,8 +1,69 @@
-use std::fmt::{Display, Formatter};
+use std::fmt::{Debug, Display, Formatter};
+use std::ops::Deref;
 use uv_normalize::{ExtraName, GroupName, PackageName};
 use uv_pep440::Version;
 use version_ranges::Ranges;
 
+/// Orphan-rule workaround for inspecting build frontend errors.
```

Comment:
Might need to rope in @BurntSushi to review this implementation detail

Reply:
One thing that would be useful here is a little more docs like, "If the orphan-rule didn't exist, we would do ... instead."